pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -1210,12 +1220,17 @@ func (c *Cloud) ensureLoadBalancer(namespacedName types.NamespacedName, loadBala
 	return loadBalancer, nil
 }
 
-func createSubnetMappings(subnetIDs []string) []*elbv2.SubnetMapping {
+func createSubnetMappings(subnetIDs []string, allocationIDs []string) []*elbv2.SubnetMapping {
 	response := []*elbv2.SubnetMapping{}
 
-	for _, id := range subnetIDs {
-		// Ignore AllocationId for now
-		response = append(response, &elbv2.SubnetMapping{SubnetId: aws.String(id)})
+	for index, id := range subnetIDs {
+		var sm *elbv2.SubnetMapping
```

Comment:
We can simplify as

```go
sm := &elbv2.SubnetMapping{SubnetId: aws.String(id)}
if len(allocationIDs) > 0 {
  sm.AllocationId = aws.String(allocationIDs[index])
}
```

Just suggestion :)

Reply:
Thank you for the suggestion. Implemented in 99a6ea0322.